qiskit/primitives/containers/data_bin.py

```diff
@@ -15,78 +15,78 @@
 """
 from __future__ import annotations
 
-from collections.abc import Iterable
-from dataclasses import make_dataclass
+from typing import Any, Iterable
 
 
 class DataBinMeta(type):
-    """Metaclass for :class:`DataBin` that adds the shape to the type name.
+    """This metaclass no longer has a purpose and will be removed."""
```

Comment:
Is there any existing code that uses this class that prevents moving it now? The only thing I can think to check is JSON serialization in the runtime

Reply:
I looked at all of our repos, public and private, and I didn't find any usage. I have therefore removed the metaclass altogether in this PR.